src/js/lib/search.js

```diff
@@ -0,0 +1,346 @@
+import bodybuilder from "bodybuilder"
+import { either, isEmpty, isNil, equals } from "ramda"
+
+import { emptyOrNil } from "./util"
+import {
+  LR_TYPE_COURSE,
+  LR_TYPE_PROGRAM,
+  LR_TYPE_USERLIST,
+  LR_TYPE_LEARNINGPATH,
+  LR_TYPE_VIDEO,
+  LR_TYPE_PODCAST,
+  LR_TYPE_PODCAST_EPISODE,
+  OCW_PLATFORM
+} from "./constants"
+
+const LEARN_SUGGEST_FIELDS = ["title.trigram", "short_description.trigram"]
+
+const OBJECT_TYPE = "type"
+
+const QUERY_FIELDS = [
+  "title.english^3",
+  "short_description.english^2",
+  "full_description.english",
+  "topics",
+  "platform",
+  "course_id",
+  "coursenum^5",
+  "offered_by"
+]
+
+export const RESOURCE_QUERY_NESTED_FIELDS = [
+  "runs.year",
+  "runs.semester",
+  "runs.level",
+  "runs.instructors^5"
+]
+
+export const LR_TYPE_ALL = [
+  LR_TYPE_COURSE,
+  LR_TYPE_PROGRAM,
+  LR_TYPE_USERLIST,
+  LR_TYPE_LEARNINGPATH,
+  LR_TYPE_VIDEO,
+  LR_TYPE_PODCAST,
+  LR_TYPE_PODCAST_EPISODE
+]
```

Comment:
You can remove `LR_TYPE_USERLIST` and `LR_TYPE_LEARNINGPATH` since those are specific to open-discussions.  Also maybe `LR_TYPE_PROGRAM`,  I don't think OCW has programs?  And add in "resourcefile" as a new LR_TYPE_RESOURCEFILE constant.